src/github.com/getlantern/flashlight/config/fetcher.go

```diff
@@ -49,8 +48,8 @@ type UserConfig interface {
 
 // NewFetcher creates a new configuration fetcher with the specified
 // interface for obtaining the user ID and token if those are populated.
-func NewFetcher(conf UserConfig, httpFetcher util.HTTPFetcher) Fetcher {
-	return &fetcher{lastCloudConfigETag: map[string]string{}, user: conf, httpFetcher: httpFetcher}
+func NewFetcher(conf UserConfig, httpClient *http.Client) Fetcher {
```

Comment:
Yeah I actually think the Fetcher (maybe there's a better name) adds a lot to testability. Is there anywhere we call anything but `Do` on that client? If not then to me the interface is the clear winner. If we do call something else on it, that might indicate the code there is too tightly coupled.

Maybe "HTTPDoer" straight from that article like:

```
type HTTPDoer interface {
    Do(*http.Request) (*http.Response, error)
}
```